bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/ImportJobFromHbaseSnapshot.java

```diff
@@ -0,0 +1,138 @@
+ *                --stagingLocation=gs://$STAGING_PATH \
+ *                --project=$PROJECT \
+ *                --bigtableInstanceId=$INSTANCE \
+ *                --bigtableTableId=$TABLE \
+ *                --gcsProject=$PROJECT \
+ *                --hbaseRootDir=gs://$HBASE_EXPORT_ROOT_PATH \
+ *                --snapshotName=$SNAPSHOT_NAME  \
+ *                --restoreDir=gs://$RESTORE_PATH
+ * </pre>
+ */
+@InternalExtensionOnly
+public class ImportJobFromHbaseSnapshot {
+  private static final Log LOG = LogFactory.getLog(ImportJobFromHbaseSnapshot.class);
+
+  public interface ImportOptions extends ImportJob.ImportOptions {
+    @Description("The GCP project id for the GCS bucket")
+    String getGcsProject();
+
+    @SuppressWarnings("unused")
+    void setGcsProject(String gcsProjectId);
+
+    @Description("The HBase root dir where HBase snapshot files resides.")
+    String getHbaseRootDir();
+
+    @SuppressWarnings("unused")
+    void setHbaseRootDir(String hbaseRootDir);
+
+    @Description("Temp location for restoring snapshots")
+    String getRestoreDir();
+
+    @SuppressWarnings("unused")
+    void setRestoreDir(String restoreDir);
+
+    @Description("Snapshot name")
+    String getSnapshotName();
+
+    @SuppressWarnings("unused")
+    void setSnapshotName(String snapshotName);
+  }
+
+  public static void main(String[] args) {
+    PipelineOptionsFactory.register(ImportOptions.class);
+
+    ImportOptions opts =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(ImportOptions.class);
+
+    LOG.info("DEBUG===> Building Pipeline");
+    Pipeline pipeline = buildPipeline(opts);
+
+    LOG.info("DEBUG===> Running Pipeline");
+    PipelineResult result = pipeline.run();
+
+    if (opts.getWait()) {
+      Utils.waitForPipelineToFinish(result);
+    }
+  }
+
+  @VisibleForTesting
+  static Pipeline buildPipeline(ImportOptions opts) {
+    Pipeline pipeline = Pipeline.create(Utils.tweakOptions(opts));
+    pipeline
+        .apply(
+            "Read from HBase Snapshot",
+            HadoopFormatIO.<ImmutableBytesWritable, Result>read()
+                .withConfiguration(
+                    new HBaseSnapshotInputConfigBuilder()
+                        .setProjectId(opts.getGcsProject())
+                        .setExportedSnapshotDir(opts.getHbaseRootDir())
+                        .setSnapshotName(opts.getSnapshotName())
+                        .setRestoreDir(opts.getRestoreDir())
+                        .build()))
+        .apply("Create Mutations", ParDo.of(new HBaseResultToMutationFn()))
+        .apply("Write to Bigtable", createSink(opts));
+
+    return pipeline;
+  }
+
+  static PTransform<PCollection<Mutation>, PDone> createSink(ImportOptions opts) {
+    CloudBigtableTableConfiguration config = TemplateUtils.BuildImportConfig(opts);
+    config =
+        config
+            .toBuilder()
+            .withConfiguration(BigtableOptionsFactory.BIGTABLE_BULK_AUTOFLUSH_MS_KEY, "0")
```

Comment:
I think the Autoflush bug is fixed. Please check with Mattie and remove this line.

Reply:
Yes, it is fixed and merged into master: https://github.com/googleapis/java-bigtable-hbase/pull/2772

Reply:
Thanks for confirming, will remove this line